### DISCUSSION TITLE
Extracting schema and indexes doubts

### DISCUSSION THREAD

**Discussion by stefanomarra** [halilcosdu/laravel-slower#20] [Q&A] [answered]
Hi everyone,
I have some doubts about the `extractIndexesAndSchemaFromRecord` inside `RecommendationService` https://github.com/halilcosdu/laravel-slower/blob/ae627f3aa1d45f43147925c9c10278d738e99e94/src/Services/RecommendationService.php#L44

The idea here is to extract the schema and indexes to be used in the chatgpt prompt request. But isn't this extracting the schema and indexes of the `slow_logs` table? Shouldn't it be extracting some imfornations from the slow query that is being tracked in the slow_log instead?

**Comment by halilcosdu** [comment, marked as answer]
Since we are using AI, AI is clever enough to understand what is happening behind the scenes. However, if you check the open PR, you will see the update. For now, I just don't want to make it too complicated. And you are correct btw.
